.circleci/config.yml

```diff
@@ -18,17 +18,24 @@ templates:
           USE_SYSTEM_LIBS: "1"
     working_directory: /go/src/github.com/DataDog/datadog-agent
   step_templates:
-    - restore_cache: &restore_source
-        keys:
-          # Cache retrieval is faster than full git checkout
-          - v2-repo-{{ .Revision }}
     - restore_cache: &restore_deps
         keys:
           # The first match will be used. Doing that so new branches
           # use master's cache but don't pollute it back.
-          - v2-godeps-{{ .Branch }}-{{ .Revision }}
-          - v2-godeps-{{ .Branch }}-
-          - v2-godeps-master-
+          #
+          # If incremental dep fails, increase the cache gen number
+          # in restore_deps AND save_deps
+          # See https://github.com/DataDog/datadog-agent/pull/2384
+          - gen3-godeps-{{ .Branch }}-{{ .Revision }}
+          - gen3-godeps-{{ .Branch }}-
+          - gen3-godeps-master-
+    - save_cache: &save_deps
+        key: gen3-godeps-{{ .Branch }}-{{ .Revision }}
+    - restore_cache: &restore_source
+        keys:
+          - v2-repo-{{ .Revision }}
```

Comment:
could we keep the comment `Cache retrieval is faster than full git checkout` here? Found it useful :)